### ISSUE TITLE
CollectionView.attributeBindings are Broken in 1.11.0

### ISSUE THREAD

**Issue by mmpestorich** [emberjs/ember.js#10792]
Using attributeBindings with a CollectionView generates the following error...

`Uncaught Error: The morph before which the new morph is to be inserted is not a child of this morph.`

Here's a jsbin illustrating the problem: http://emberjs.jsbin.com/wuqodo/3/

In this case it occurs when an item is added to the `CollectionView` after it initially renders while `ensureChildrenAreInDOM` is processing the `AttrNode` inside `Morph.prototype.insertBeforeMorph`. The `referenceMorph.parentMorph` is `null` and I assume that is should not be...

This might be related to issue #10601.

**Comment by stefanpenner** [comment]
I also ran into similar issues with list-view, as such I am dubious of attrNodes living `childViews`. Likely my issue was [simple to work around](https://github.com/embersherpa/list-view/commit/e29b2074ba1120fd0c27faec2f9cb1ebb0d99a5e)

@mixonic  is `ensureChildrenAreInDOM` also to ensure `AttrNodes` at in DOM? Basically, can I push an AttrNode onto a container/collection view, and attributes are updated accordingly?

**Comment by taras** [comment]
The except is thrown because `referenceMorph.parentMorph` is undefined.

**Comment by krisselden** [comment]
@mixonic we should move these from _childViews to _attrNodes and just change the Renderer to concat them with _childViews in the childViews hook.

```js
EmberRenderer.prototype.childViews = function childViews(view) {
  return view.attrNodes.concat(view._childViews);
};
```

**Comment by taras** [comment]
@krisselden @rwjblue I added `view._attrNodes` as per suggestion.

**Comment by mixonic** [comment]
@krisselden ah, that may be enough to make clean up easy. Will revisit.

**Comment by mmpestorich** [comment]
Fixed by #10795
